slatedb/src/db/builder.rs

```diff
@@ -124,7 +125,8 @@ pub struct DbBuilder<P: Into<Path>> {
     main_object_store: Arc<dyn ObjectStore>,
     wal_object_store: Option<Arc<dyn ObjectStore>>,
     block_cache: Option<Arc<dyn DbCache>>,
-    clock: Option<Arc<dyn Clock + Send + Sync>>,
+    user_clock: Option<Arc<dyn Clock>>,
+    system_clock: Option<Arc<dyn Clock>>,
```

Comment:
What I had meant by the previous comment was to use two different traits for user clock and system clock, otherwise you can accidentally pass the opposite impl into the user/system slots.